File: gowrish/graphs/dfs/dfs.cpp
#include <bits/stdc++.h>
using namespace std;

void dfs(int node, vector<int> adj[], vector<bool> &visited)
{
    visited[node] = true;
    for (int neighbor : adj[node])
    {
        if (!visited[neighbor])
        {
            dfs(neighbor, adj, visited);
        }
    }
}

int main()
{
    int n, m;
    cin >> n >> m;
    vector<int> adj[n + 1];
    for (int i = 0; i < m; i++)
    {
        int u, v;
        cin >> u >> v;
        adj[u].push_back(v);
        adj[v].push_back(u);
    }
    vector<bool> visited(n + 1, false);
    for (int i = 1; i <= n; i++)
    {
        if (!visited[i])
        {
            dfs(i, adj, visited);
        }
    }
    return 0;
}

File: gowrish/graphs/dfs/dfs.py
def dfs(node,adj,visited):
    visited[node]=True
    for i in adj[node]:
        if not visited[i]:
            dfs(i,adj,visited)
            
def main():
    n,m=map(int,input().split())
    adj=[[] for i in range(n)]
    for i in range(m):
        u,v=map(int,input().split())
        adj[u].append(v)
        adj[v].append(u)
    visited=[False]*n
    for i in range(n):
        if not visited[i]:
            dfs(i,adj,visited)
    print(visited)